api/git/git.go

```diff
@@ -19,21 +21,27 @@ func NewService(dataStorePath string) (*Service, error) {
 
 // ClonePublicRepository clones a public git repository using the specified URL in the specified
 // destination folder.
-func (service *Service) ClonePublicRepository(repositoryURL, destination string) error {
-	return cloneRepository(repositoryURL, destination)
+func (service *Service) ClonePublicRepository(repositoryURL, referenceName string, destination string) error {
+	return cloneRepository(repositoryURL, referenceName, destination)
 }
 
 // ClonePrivateRepositoryWithBasicAuth clones a private git repository using the specified URL in the specified
 // destination folder. It will use the specified username and password for basic HTTP authentication.
-func (service *Service) ClonePrivateRepositoryWithBasicAuth(repositoryURL, destination, username, password string) error {
+func (service *Service) ClonePrivateRepositoryWithBasicAuth(repositoryURL, referenceName string, destination, username, password string) error {
 	credentials := username + ":" + url.PathEscape(password)
 	repositoryURL = strings.Replace(repositoryURL, "://", "://"+credentials+"@", 1)
-	return cloneRepository(repositoryURL, destination)
+	return cloneRepository(repositoryURL, referenceName, destination)
 }
 
-func cloneRepository(repositoryURL, destination string) error {
-	_, err := git.PlainClone(destination, false, &git.CloneOptions{
+func cloneRepository(repositoryURL, referenceName string, destination string) error {
+	stack := &git.CloneOptions{
 		URL: repositoryURL,
-	})
+	}
+
+	if !govalidator.IsNull(referenceName) {
```

Comment:
The govalidator package should only be used when validating HTTP requests payload. Instead, please check against the default string value `""`:

```go
if referenceName != "" {
  options.ReferenceName = plumbing.ReferenceName(referenceName)
}
 ```